### PR TITLE
feat!: move from path strings to URLs

### PR DESCRIPTION
**Pull request by blacha**
#### Motivation

Switching to URLS allows us to be more speicifc and consistent with paths, especially when working between s3, http, windows, mac and linux file systems.

It is especially helpful with files vs folders as URLs need folders to end with "/".

#### Modification

Upgrades chunkd/aws-sdk/cogeotiff to latest versions 
Switches most file access to URL based

#### Checklist

_If not applicable, provide explanation of why._

- [ ] Tests updated
- [ ] Docs updated
- [ ] Issue linked in Title

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -40,16 +40,15 @@
...
     "@basemaps/server": "^7.0.0",
     "@basemaps/shared": "^7.0.0",
     "@basemaps/tiler": "^7.0.0",
     "@basemaps/tiler-sharp": "^7.0.0",
-    "@chunkd/fs": "^10.0.2",
-    "@chunkd/source-aws-v2": "^9.3.1",
-    "@cogeotiff/core": "^7.2.0",
-    "@cotar/core": "^5.4.0",
-    "@cotar/tar": "^5.4.1",
+    "@chunkd/source-memory": "^11.0.2",
```
Comment (Wentao-Kuang) [inline packages/cli/package.json:49]: I think all `@chunkd` library are from `shared` package now?

Comment (blacha) [reply packages/cli/package.json:49]: the CLI needs more packages than what shared provides.

File: packages/cli/src/cli/config/action.bundle.assets.ts [hunk 2]
```diff
@@ -65,20 +63,20 @@ export class CommandBundleAssets extends CommandLineAction {
...
     const tarBuilder = new TarBuilder(outputTar);
 
     const files = await fsa.toArray(fsa.list(input));
     // Ensure files are put into the same order
-    files.sort((a, b) => a.localeCompare(b));
+    files.sort((a, b) => a.href.localeCompare(b.href));
     logger.info({ output: outputTar, files: files.length }, 'Tar:Create');
 
-    const basePath = path.resolve(input);
+    // const basePath = path.resolve(input);
```
Comment (Wentao-Kuang) [inline packages/cli/src/cli/config/action.bundle.assets.ts:77]:
```suggestion
```

File: packages/config-loader/src/json/json.config.ts [hunk 3]
```diff
@@ -331,7 +340,8 @@ export class ConfigJson {
       updatedAt: now,
       projection: tileMatrix.projection.code,
       tileMatrix: tileMatrix.identifier,
-      uri,
+      uri: url.href,
+      // url,
```
Comment (Wentao-Kuang) [inline packages/config-loader/src/json/json.config.ts:344]:
```suggestion
```

File: packages/lambda-tiler/package.json [hunk 4]
```diff
@@ -50,8 +48,8 @@
     "bundle": "./bundle.sh"
   },
   "devDependencies": {
+    "@chunkd/fs": "^11.2.0",
```
Comment (Wentao-Kuang) [inline packages/lambda-tiler/package.json:51]: Do we still need this here?

File: packages/lambda-tiler/src/__tests__/config.data.ts [hunk 5]
```diff
@@ -277,7 +277,8 @@ export class FakeData {
     const fsMemory = new FsMemory();
 
     const configPath = `memory://linz-basemaps/${output.hash}.json`;
-    fsMemory.files.set(configPath, Buffer.from(JSON.stringify(output)));
+    console.log(configPath);
```
Comment (Wentao-Kuang) [inline packages/lambda-tiler/src/__tests__/config.data.ts:280]:
```suggestion
```

File: packages/lambda-tiler/src/__tests__/config.data.ts [hunk 6]
```diff
@@ -9,8 +9,8 @@ import {
   TileSetType,
 } from '@basemaps/config';
 import { ImageFormat, VectorFormat } from '@basemaps/geo';
-import { fsa } from '@basemaps/shared';
-import { FsMemory } from '@chunkd/source-memory';
+import { fsa, FsMemory } from '@basemaps/shared';
+// import { FsMemory } from '@chunkd/fs';
```
Comment (Wentao-Kuang) [inline packages/lambda-tiler/src/__tests__/config.data.ts:13]:
```suggestion
```

File: packages/lambda-tiler/package.json (hunk 4 above)
Comment (blacha) [reply packages/lambda-tiler/package.json:51]: Yeah its used in the tests :(

**Review by Wentao-Kuang — APPROVED** [review]

**Pull request merged**